subsys/usb/device_next/class/usbd_midi.c

```diff
@@ -0,0 +1,750 @@
+		},                                                                       \
+		.if1_1_cs_in_ep = {                                                      \
+			.bLength = 4 + N_INPUTS(inst),                                   \
+			.bDescriptorType = CS_ENDPOINT,                                  \
+			.bDescriptorSubtype = MS_GENERAL_2_0,                            \
+			.bNumGrpTrmBlock = N_INPUTS(inst),                               \
+			.baAssoGrpTrmBlkID = {GRPTRM_INPUT_BLOCK_IDS(inst)},             \
+		},                                                                       \
+		.grptrm_header = {                                                       \
+			.bLength = sizeof(struct usb_midi_grptrm_header_descriptor),     \
+			.bDescriptorType = CS_GR_TRM_BLOCK,                              \
+			.bDescriptorSubtype = GR_TRM_BLOCK_HEADER,                       \
+			.wTotalLength = sys_cpu_to_le16(                                 \
+				USBD_MIDI2_GRPTRM_TOTAL_LEN(inst)                        \
+			),                                                               \
+		},                                                                       \
+		.grptrm_blocks = {                                                       \
+			DT_INST_FOREACH_CHILD_SEP(                                       \
+				inst, USBD_MIDI2_INIT_GRPTRM_BLOCK_DESCRIPTOR, (,)       \
+			)                                                                \
+		},                                                                       \
+	};                                                                               \
+	static const struct usb_desc_header *usbd_midi_desc_array_fs_##inst[] = {        \
+		(struct usb_desc_header *)&usbd_midi_desc_##inst.iad,                    \
+		(struct usb_desc_header *)&usbd_midi_desc_##inst.if0_std,                \
+		(struct usb_desc_header *)&usbd_midi_desc_##inst.if0_cs,                 \
+		(struct usb_desc_header *)&usbd_midi_desc_##inst.if1_0_std,              \
+		(struct usb_desc_header *)&usbd_midi_desc_##inst.if1_0_ms_header,        \
+		(struct usb_desc_header *)&usbd_midi_desc_##inst.if1_0_out_ep_fs,        \
+		(struct usb_desc_header *)&usbd_midi_desc_##inst.if1_0_cs_out_ep,        \
+		(struct usb_desc_header *)&usbd_midi_desc_##inst.if1_0_in_ep_fs,         \
+		(struct usb_desc_header *)&usbd_midi_desc_##inst.if1_0_cs_in_ep,         \
+		(struct usb_desc_header *)&usbd_midi_desc_##inst.if1_1_std,              \
+		(struct usb_desc_header *)&usbd_midi_desc_##inst.if1_1_ms_header,        \
+		(struct usb_desc_header *)&usbd_midi_desc_##inst.if1_1_out_ep_fs,        \
+		(struct usb_desc_header *)&usbd_midi_desc_##inst.if1_1_cs_out_ep,        \
+		(struct usb_desc_header *)&usbd_midi_desc_##inst.if1_1_in_ep_fs,         \
+		(struct usb_desc_header *)&usbd_midi_desc_##inst.if1_1_cs_in_ep,         \
+	};                                                                               \
```

Comment:
Could you please check if a termination of the array is missing, either a NULL or a nil-descriptor (like in e.g. usbd_cdc_acm.c)? init_configuration_inst checks for that condition (https://github.com/zephyrproject-rtos/zephyr/blob/6f240ef18c11473d05aca9954c9ef0d91a19d7db/subsys/usb/device_next/usbd_init.c#L135)

Reply:
Oh that's a very good catch :+1: 

I hadn't noticed that the CDC-ACM nil descriptor was "special", and I was wondering by which magic the end of array was correctly detected by usbd.

Reply:
Maybe that caused the issue "It does not work, I guess there is something wrong in the interface descriptors." that @jfischer-no reported?